internal/resolution/variable_sources/entity/bundle_entity.go

```diff
@@ -212,6 +203,19 @@ func (b *BundleEntity) loadChannelProperties() error {
 	return nil
 }
 
+func (b *BundleEntity) loadBundlePath() error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	if b.bundlePath == "" {
+		bundlePath, err := loadFromEntity[string](b.Entity, PropertyBundlePath, required)
```

Comment:
I'm not sure that this is necessary. From my testing, this particular line is what makes us have to do weird marshalling and unmarshalling logic [here](https://github.com/operator-framework/operator-controller/pull/182/files#diff-1b647d2198aaf1054b9a90ee4b1ec25fbcc7ed2b7cd68d63e6a89008a3dfe6f8R92-R96)

I think using the `loadFromEntity()` function makes the most sense when you are attempting to load something you are expecting to follow a specific structure and gets unmarshalled into a `struct`. For this case, since we are expecting an unstructured `string` (`"hello world!"` as opposed to `"{"test": "test"}"`) and the `b.Entity.Properties` is a `map[string]string` I think we are better off just doing something like:
```suggestion
		bundlePath, ok := b.Entity.Properties[PropertyBundlePath]
```

Using the above suggestion will allow us to change [this](https://github.com/operator-framework/operator-controller/pull/182/files#diff-1b647d2198aaf1054b9a90ee4b1ec25fbcc7ed2b7cd68d63e6a89008a3dfe6f8R92-R96) to:
```go
props["olm.bundle.path"] = bundle.Spec.Image
```